### ISSUE TITLE
how to make sure that the processed data are consistent

### ISSUE THREAD

**Issue by k2s** [datproject/dat-node#68]
I need to download data from live source and make sure that long running `processConsistentData()` will work with not changed data.
Are there some `pause()`/`resume()` functions ? Best would be to continue distribution of data (upload) and only stop synchronization with incomming changes (`pauseDownload()`/`resumeDownload()`).

```js
Dat(targetFolder, { key: downKey }, function (err, dat) {
      if (err) {
        return console.error(err)
      }

      // Join the network
      // var network =
      dat.joinNetwork(/*opts*/)
      // network.swarm // hyperdiscovery
      // network.connected // number of connected peers

      // Track stats
      var stats = dat.trackStats() // hyperdrive-stats
      // stats.network // hyperdrive-network-speed

      stats.on('update', function () {
        let p = stats.get()
        if (p.blocksProgress !== p.blocksTotal) return
        // this.pause() or this.pauseDownload()
        processConsistentData()
        // this.resume() or this.resumeDownload()
      })
})
```

**Comment by joehand** [comment]
Hello! There is not a great way to do `pause` and `resume` right now. But we need to add it!

Right now, the most straightforward way would be to leave the network on the download side once the initial download is complete:

```js
// Inside the callback
var network = dat.joinNetwork()

network.swarm.leave(dat.archive.discoveryKey)
```

Then to resume, you have to rejoin the network:

```js
network.swarm.join(dat.archive.discoveryKey)
```

The downside is that you have to rediscover the peers each time you leave/join.

We'll be adding [join](https://github.com/datproject/dat-node/blob/multidrive-support/dat.js#L31) and [leave](https://github.com/datproject/dat-node/blob/multidrive-support/dat.js#L50) in the next release of dat-node.

The other option would involve calling `dat.archive.unreplicate()` ([docs here](https://github.com/mafintosh/hyperdrive#archiveunreplicatestream)). This would pause the download, but it's not clear to me how you'd resume it easily.

**Comment by joehand** [comment]
@k2s added a new issue for pause/resume! https://github.com/datproject/dat-node/issues/69 Feel free to subscribe there if you'd like to track that. 

That issue also has another implementation you could use that I hadn't thought of earlier.

**Comment by k2s** [comment]
@joehand thank you for quick answer

closing in favor of #69
